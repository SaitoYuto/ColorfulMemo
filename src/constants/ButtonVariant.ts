export const BUTTON_VARIANT = {
  TEXT: "text",
  TONAL: "tonal",
  FLAT: "flat",
  ELEVATED: "elevated",
  OUTLINED: "outlined",
  PLAIN: "plain",
  UNDEFINED: undefined,
} as const;
