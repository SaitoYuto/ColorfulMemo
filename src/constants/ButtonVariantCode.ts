export const BUTTON_VARIANT_CODE = {
  TEXT: 1,
  TONAL: 2,
  FLAT: 3,
  ELEVATED: 4,
  OUTLINED: 5,
  PLAIN: 6,
} as const;
