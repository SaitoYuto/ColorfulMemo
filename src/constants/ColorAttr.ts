export const COLOR_ATTR = {
  DEFAULT: {
    ALERT: "#757575",
    BAR: "#9E9E9E",
    BUTTON: "#9E9E9E",
    DIVIDER: "#757575",
    ICON: "#9E9E9E",
    LIST: "#757575",
    TEXT_FIELD: "#F5F5F5",
    TEXT_AREA: "#F5F5F5",
    SELECT: "#F5F5F5",
  } as const,
  BLUE: {
    ALERT: "rgb(33, 150, 243)",
    BAR: "#2196F3",
    BUTTON: "#2196F3",
    DIVIDER: "rgb(33, 150, 243)",
    ICON: "#2196F3",
    LIST: "rgb(33, 150, 243)",
    TEXT_FIELD: "#BBDEFB",
    TEXT_AREA: "#BBDEFB",
    SELECT: "#BBDEFB",
  } as const,
  RED: {
    ALERT: "rgb(176, 0, 32)",
    BAR: "#F44336",
    BUTTON: "#F44336",
    DIVIDER: "rgb(176, 0, 32)",
    ICON: "#F44336",
    LIST: "rgb(176, 0, 32)",
    TEXT_FIELD: "#FFCDD2",
    TEXT_AREA: "#FFCDD2",
    SELECT: "#FFCDD2",
  } as const,
  YELLOW: {
    ALERT: "rgb(251, 140, 0)",
    BAR: "#FFEB3B",
    BUTTON: "#FFEB3B",
    DIVIDER: "rgb(251, 140, 0)",
    ICON: "#FFEB3B",
    LIST: "rgb(251, 140, 0)",
    TEXT_FIELD: "#FFF9C4",
    TEXT_AREA: "#FFF9C4",
    SELECT: "#FFF9C4",
  } as const,
  GREEN: {
    ALERT: "rgb(76, 175, 80)",
    BAR: "#4CAF50",
    BUTTON: "#4CAF50",
    DIVIDER: "rgb(76, 175, 80)",
    ICON: "#4CAF50",
    LIST: "rgb(76, 175, 80)",
    TEXT_FIELD: "#C8E6C9",
    TEXT_AREA: "#C8E6C9",
    SELECT: "#C8E6C9",
  } as const,
} as const;
