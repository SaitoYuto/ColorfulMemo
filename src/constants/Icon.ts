export const ICON = {
  MENU: "mdi-menu",
  HOME: "mdi-home",
  PENCIL: "mdi-pencil",
  WRENCH: "mdi-wrench",
  GITHUB: "mdi-github",
} as const;
