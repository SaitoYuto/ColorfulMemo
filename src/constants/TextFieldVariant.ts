export const TEXTFIELD_VARIANT = {
  FILLED: "filled",
  UNDERLINED: "underlined",
  OUTLINED: "outlined",
  PLAIN: "plain",
  SOLO: "solo",
  SOLO_INVERTED: "solo-inverted",
  SOLO_FILLED: "solo-filled",
  UNDEFINED: undefined,
} as const;
