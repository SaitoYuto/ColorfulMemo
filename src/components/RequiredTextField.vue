<template>
  <v-text-field
    :bg-color="bgColor"
    :class="$props.class"
    :clearable="clearable"
    :label="label"
    :rows="rows"
    :rules="[isRequired]"
    :variant="variant"
    v-model="input"
  ></v-text-field>
</template>

<script lang="ts" setup>
import { MESSAGE } from "@/constants/Message";
import { TEXTFIELD_VARIANT } from "@/constants/TextFieldVariant";
import type { TextFieldVariant } from "@/types/TextFieldVariant";

defineProps({
  bgColor: String,
  clearable: {
    type: Boolean,
    default: true,
  },
  label: String,
  variant: {
    type: String as PropType<TextFieldVariant>,
    default: TEXTFIELD_VARIANT.OUTLINED,
  },
  rows: {
    type: String,
    default: "1",
  },
  class: {
    type: String,
    default: "mt-4",
  },
});

const input = defineModel("input", {
  type: String,
  default: "",
});

/**
 * check if the input exists
 *
 * @returns {boolean | string}
 */
function isRequired(): boolean | string {
  return input.value.length > 0 || MESSAGE.FIELD.REQUIRED;
}
</script>
