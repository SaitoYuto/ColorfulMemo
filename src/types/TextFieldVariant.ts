import { TEXTFIELD_VARIANT } from "@/constants/TextFieldVariant";

export type TextFieldVariant =
  (typeof TEXTFIELD_VARIANT)[keyof typeof TEXTFIELD_VARIANT];
