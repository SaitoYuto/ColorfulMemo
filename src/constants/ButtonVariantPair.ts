import { BUTTON_VARIANT } from "@/constants/ButtonVariant";
import { CodeText } from "@/classes/CodeText";
import { BUTTON_VARIANT_CODE } from "@/constants/ButtonVariantCode";
import { ButtonVariantCode } from "@/types/ButtonVariantCode";

export const BUTTON_VARIANT_PAIR: CodeText<ButtonVariantCode> =
  new (class extends CodeText<ButtonVariantCode> {
    protected setup(): void {
      this.append(BUTTON_VARIANT_CODE.TEXT, BUTTON_VARIANT.TEXT);
      this.append(BUTTON_VARIANT_CODE.TONAL, BUTTON_VARIANT.TONAL);
      this.append(BUTTON_VARIANT_CODE.FLAT, BUTTON_VARIANT.FLAT);
      this.append(BUTTON_VARIANT_CODE.ELEVATED, BUTTON_VARIANT.ELEVATED);
      this.append(BUTTON_VARIANT_CODE.OUTLINED, BUTTON_VARIANT.OUTLINED);
      this.append(BUTTON_VARIANT_CODE.PLAIN, BUTTON_VARIANT.PLAIN);
    }
  })();
