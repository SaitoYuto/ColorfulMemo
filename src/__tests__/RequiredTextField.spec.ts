import RequiredTextField from "@/components/RequiredTextField.vue";
import { mount } from "@vue/test-utils";
import "@/__tests__/setup";
import { TEXTFIELD_VARIANT } from "@/constants/TextFieldVariant";

describe("RequiredTextField.vue", () => {
  it("renders correctly with default props", () => {
    const wrapper = mount(RequiredTextField);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders with provided props", () => {
    const wrapper = mount(RequiredTextField, {
      props: {
        bgColor: "red",
        clearable: false,
        label: "Test Label",
        variant: "filled",
        rows: "1",
        class: "custom-class",
      },
    });

    expect(wrapper.props("bgColor")).toBe("red");
    expect(wrapper.props("clearable")).toBe(false);
    expect(wrapper.props("label")).toBe("Test Label");
    expect(wrapper.props("variant")).toBe(TEXTFIELD_VARIANT.FILLED);
    expect(wrapper.props("class")).toBe("custom-class");
  });
});
