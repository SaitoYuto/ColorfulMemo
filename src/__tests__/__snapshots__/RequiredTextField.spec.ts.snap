// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RequiredTextField.vue > renders correctly with default props 1`] = `
"<div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field mt-4">
  <!---->
  <div class="v-input__control">
    <div class="v-field v-field--appended v-field--center-affix v-field--no-label v-field--variant-outlined v-theme--light v-locale--is-ltr">
      <div class="v-field__overlay"></div>
      <div class="v-field__loader">
        <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
          <!---->
          <div class="v-progress-linear__background" style="width: 100%;"></div>
          <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
            <div class="v-progress-linear__indeterminate">
              <div class="v-progress-linear__indeterminate long"></div>
              <div class="v-progress-linear__indeterminate short"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <div class="v-field__field" data-no-activator="">
        <!----><label class="v-label v-field-label" for="input-0">
          <!---->
          <!---->
        </label>
        <!----><input size="1" type="text" id="input-0" aria-describedby="input-0-messages" rows="1" class="v-field__input">
        <!---->
      </div>
      <transition-stub name="expand-x-transition" appear="false" persisted="false" css="true">
        <div class="v-field__clearable" style="display: none;"><i class="mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" aria-label="Clear "></i></div>
      </transition-stub>
      <div class="v-field__append-inner">
        <!---->
        <!---->
      </div>
      <div class="v-field__outline">
        <div class="v-field__outline__start"></div>
        <!---->
        <div class="v-field__outline__end"></div>
        <!---->
      </div>
    </div>
  </div>
  <!---->
  <div class="v-input__details">
    <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-0-messages">
      <!---->
    </transition-group-stub>
    <!---->
  </div>
</div>"
`;
